src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/KeyworkerServiceController.kt

```diff
@@ -0,0 +1,291 @@
+import springfox.documentation.annotations.ApiIgnore
+import uk.gov.justice.digital.hmpps.keyworker.dto.AllocationsFilterDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.BasicKeyworkerDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.ErrorResponse
+import uk.gov.justice.digital.hmpps.keyworker.dto.KeyworkerAllocationDetailsDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.KeyworkerAllocationDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.KeyworkerDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.KeyworkerUpdateDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.OffenderKeyWorkerHistory
+import uk.gov.justice.digital.hmpps.keyworker.dto.OffenderKeyworkerDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.OffenderLocationDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.PagingAndSortingDto
+import uk.gov.justice.digital.hmpps.keyworker.dto.Prison
+import uk.gov.justice.digital.hmpps.keyworker.dto.SortOrder
+import uk.gov.justice.digital.hmpps.keyworker.model.AllocationType
+import uk.gov.justice.digital.hmpps.keyworker.model.KeyworkerStatus
+import uk.gov.justice.digital.hmpps.keyworker.rolemigration.UserRolesMigrationService
+import uk.gov.justice.digital.hmpps.keyworker.services.KeyworkerAutoAllocationService
+import uk.gov.justice.digital.hmpps.keyworker.services.KeyworkerMigrationService
+import uk.gov.justice.digital.hmpps.keyworker.services.KeyworkerService
+import uk.gov.justice.digital.hmpps.keyworker.services.PrisonSupportedService
+import java.time.LocalDate
+import java.util.Optional
+import javax.persistence.EntityNotFoundException
+import javax.validation.Valid
+
+@Api(tags = ["key-worker"])
+@RestController
+@RequestMapping(value = ["key-worker"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class KeyworkerServiceController(
+    private val keyworkerService: KeyworkerService,
+    private val keyworkerMigrationService: KeyworkerMigrationService,
+    private val keyworkerAutoAllocationService: KeyworkerAutoAllocationService,
+    private val roleMigrationService: UserRolesMigrationService,
+    private val prisonSupportedService: PrisonSupportedService
+) {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(this::class.java)
+    }
+
+    @ApiOperation(value = "Key workers available for allocation at specified prison.", notes = "Key workers available for allocation at specified prison.", nickname = "getAvailableKeyworkers")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "OK", response = KeyworkerDto::class, responseContainer = "List"), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class, responseContainer = "List"), ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse::class, responseContainer = "List"), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class, responseContainer = "List")])
```

Comment:
Worth checking but I don't need the default `ApiResponse(code = 200, message = "OK", response = KeyworkerDto::class) ` response is needed as it's generated by default.

Reply:
agreed, not required, not sure I'd fix as part of the conversion to kotlin though